runtime/expr/values.go

```diff
@@ -178,95 +178,63 @@ func (r *recordSpreadExpr) invalidate(object map[string]column) {
 }
 
 type ArrayExpr struct {
-	zctx    *zed.Context
-	typ     *zed.TypeArray
-	builder *zcode.Builder
+	builder zcode.Builder
+	elems   collectionBuilder
 	exprs   []Evaluator
+	zctx    *zed.Context
```

Comment:
No need to adopt this if it doesn't warm your heart, but I find it helpful to group fields into stuff from the outside world that was passed to the constructor followed by a separate group of the purely internal stuff that's either created by the constructor or just left as the zero value.
```suggestion
	exprs []Evaluator
	zctx  *zed.Context

	builder zcode.Builder
	elems   collectionBuilder
```